apps/src/code-studio/initApp/loadApp.js

```diff
@@ -276,13 +276,51 @@ function tryToUploadShareImageToS3({image, level}) {
 }
 
 /**
+ * Loads project and checks to see if it is abusive.
+ * @returns {Promise.<AppOptionsConfig>} Resolves when project has loaded and is
+ * not abusive. Never resolves if abusive.
+ */
+function loadProjectAndCheckAbuse(appOptions) {
+  return new Promise((resolve, reject) => {
+    project.load().then(() => {
+      if (project.hideBecauseAbusive() && !appOptions.canResetAbuse) {
+        renderAbusive(window.dashboard.i18n.t('project.abuse.tos'));
+        return;
+      }
+      if (project.hideBecausePrivacyViolationOrProfane()) {
+        renderAbusive(window.dashboard.i18n.t('project.abuse.policy_violation'));
+        return;
+      }
+      resolve(appOptions);
+    });
+  });
+}
+
+/**
  * @param {AppOptionsConfig} appOptions
  * @return {Promise.<AppOptionsConfig>}
  */
 function loadAppAsync(appOptions) {
-  return new Promise((resolve, reject) => {
-    setupApp(appOptions);
+  setupApp(appOptions);
```

Comment:
These changes are easier to understand by ignoring whitespace https://github.com/code-dot-org/code-dot-org/pull/17864/files?w=1